HingeAngle/src/main/java/com/microsoft/device/display/samples/hingeangle/MainActivity.kt

```diff
@@ -7,45 +7,76 @@
 
 package com.microsoft.device.display.samples.hingeangle
 
+import android.content.res.Configuration
 import android.os.Bundle
 import android.util.Log
 import androidx.appcompat.app.AppCompatActivity
-import com.microsoft.device.dualscreen.layout.ScreenHelper
+import com.microsoft.device.dualscreen.ScreenInfo
+import com.microsoft.device.dualscreen.ScreenInfoListener
+import com.microsoft.device.dualscreen.ScreenManagerProvider
 
-class MainActivity : AppCompatActivity() {
+class MainActivity : AppCompatActivity(), ScreenInfoListener {
 
     private val DEBUG_TAG = "MainActivity"
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        ScreenManagerProvider.getScreenManager().addScreenInfoListener(this)
+    }
+
+    override fun onPause() {
+        super.onPause()
+        ScreenManagerProvider.getScreenManager().removeScreenInfoListener(this)
+    }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {
+        super.onConfigurationChanged(newConfig)
+        ScreenManagerProvider.getScreenManager().onConfigurationChanged()
+    }
+
+    override fun onScreenInfoChanged(screenInfo: ScreenInfo) {
+        setupLayout(screenInfo)
+    }
 
-        if (ScreenHelper.isDualMode(this)) {
-            Log.d(DEBUG_TAG, "########### dual mode")
+    private fun setupLayout(screenInfo: ScreenInfo) {
+        if (screenInfo.isDualMode()) {
+            useDualMode(screenInfo)
+        } else {
+            useSingleMode()
+        }
+    }
 
-            supportFragmentManager
+    private fun useDualMode(screenInfo: ScreenInfo) {
+        Log.d(DEBUG_TAG, "########### dual mode")
+        supportFragmentManager
                 .beginTransaction()
                 .replace(
-                    R.id.dual_screen_start_container_id,
-                    SingleScreenFragment(),
-                    null
+                        R.id.first_container_id,
+                        SingleScreenFragment(),
+                        null
                 ).commit()
-            supportFragmentManager
+        supportFragmentManager
                 .beginTransaction()
                 .replace(
-                    R.id.dual_screen_end_container_id,
-                    DualScreenFragment(),
-                    null
+                        R.id.second_container_id,
+                        DualScreenFragment(),
+                        null
                 ).commit()
-        } else {
-            Log.d(DEBUG_TAG, "########### single mode")
-            supportFragmentManager
+    }
+
+    private fun useSingleMode() { // Setting layout for single portrait
```

Comment:
Just for single portrait? and single landscape? If it's for both cases probably good to remove the comment that can cause confusion.